WcaOnRails/app/helpers/notifications_helper.rb

```diff
@@ -68,13 +68,24 @@ def notifications_for_user(user)
     end
 
     user.delegated_competitions.visible.over
-        .includes(:delegate_report).where(delegate_reports: { posted_at: nil })
-        .each do |competition|
+      .each do |competition|
```

Comment:
I'm pretty sure that removing the `.includes(:delegate_report)` introduces an N+1 query here (see https://secure.phabricator.com/book/phabcontrib/article/n_plus_one/ for an explanation).

Reply:
If I am using .includes(:delegate_report) then I get some notification:

user: lf
AVOID eager loading detected
  Competition => [:delegate_report]
  Remove from your finder: :includes => [:delegate_report]
[...]

In my noobish imagination I assumed this is worse than going without .includes(:delegate_report) which does not provide any messages. Am I wrong?

Reply:
Damn, what a pain =( We have this tool (bullet) that tries to autodetect possible N+1 issues, but I'm not sure how accurate it actually is.

When running the server, you should see a list of all SQL queries that get generated when viewing a page. Can you try both with and without the `.includes` and see which queries end up getting generated? Hopefully it will be straightforward from looking at that which is the right way to go. If it turns out that `bullet` is wrong, we can teach it to ignore this query (see https://github.com/jfly/worldcubeassociation.org/blob/866eab8597685c45fb06d6847c0a118ec84b662f/WcaOnRails/config/environments/development.rb#L76-L82)

Reply:
OK I've looked through the requests and in both ways actually N+1 queries are getting generated.. :/ I assume this is not created by removing `.includes(:delegate_report)` but using if-else cases what calls each competition separate anyway. 

I can avoid this by looking at results and delegate reports in *two separate loops* but I would have liked to do it in one run.